docs/errors_overview.md

```diff
@@ -0,0 +1,36 @@
+# XLA Errors Overview
+
+XLA errors are categorized into different XLA error sources. Each source has
+a list of an additional context other than the error message, which will be
+attached to each error within the category.
+
+🚧 Please note that this standarization effort is a work in progress so not all
```

Comment:
Style notice: Don't use the term "Please note".

Reference URL: https://developers.google.com/style/word-list#please